src/amp-story-embed.js

```diff
@@ -123,23 +211,32 @@ export class AmpStoryEmbed {
   }
 
   /**
-   * @return {!Promise}
    * @public
    */
   layoutCallback() {
-    // TODO: Layout all child iframes.
-    return this.layoutIframe_(0);
+    if (this.isLaidOut_) {
+      return;
+    }
+
+    const iframes = toArray(this.rootEl_.querySelectorAll('iframe'));
+    this.stories_.forEach((story, idx) => {
+      this.layoutIframe_(story, iframes[idx]);
+    });
+    this.isLaidOut_ = true;
   }
 
   /**
-   * @param {number} index
-   * @return {!Promise}
+   * @param {!Element} story
+   * @param {!Element} iframe
    * @private
    */
-  layoutIframe_(index) {
-    const story = this.stories_[index];
-    this.iframeEl_.setAttribute('src', story.href);
-    return Promise.resolve();
+  layoutIframe_(story, iframe) {
```

Comment:
The expected behavior for this library is:
1. Load one story, prerender the next story, do nothing with all the other ones
2. When navigating to the next story, prerender the following one

This PR/method is setting you on the wrong track, having a mechanism that loads one story is closer to the end goal than a mechanism that loads all the stories, as you can re-use the code to load the n+1 story.